server/src-lib/Hasura/RQL/DML/Select/Internal.hs

```diff
@@ -153,6 +152,20 @@ buildJsonObject pfx parAls arrRelCtx flds =
                      ANIField (fldAls, arrSel)
         in S.mkQIdenExp arrPfx fldAls
 
+    toSQLCol :: PGColInfo -> S.SQLExp
+    toSQLCol col = toJSONableExp (pgiType col) $
+                    S.mkQIdenExp (mkBaseTableAls pfx) $ pgiName col
+
```

Comment:
Merge `toSQLCol` and `toSQLColArgs`.

```haskell
toSQLCol col colOpM =
  toJSONableExp (pgiType col) $ case colOpM of
      Nothing -> colNameExp
      Just (ColOp op cExp) -> S.mkSQLOpExp op colNameExp cExp
  where
     colNameExp = S.mkQIdenExp (mkBaseTableAls pfx) $ pgiName col
```
This code looks precise. `toJSONableExp` logic is not redudant.

Note:-
  Wherever you feel you're repeating a logic (piece of code), try to unify them.